File: 2089-maximum-matrix-sum/maximum-matrix-sum.ts
function maxMatrixSum(matrix: number[][]): number {
    let sum = 0, isEvenNegCount = true, smallestPos = Number.MAX_SAFE_INTEGER;
    for(let i = 0; i < matrix.length; i++) {
        for(let k = 0; k < matrix[0].length; k++) {
            if(matrix[i][k] < 0) isEvenNegCount = !isEvenNegCount
            const absValue = Math.abs(matrix[i][k])
            sum += absValue;
            smallestPos = Math.min(smallestPos, absValue)
        }
    }
    return isEvenNegCount ? sum : sum - (2 * smallestPos)
};

File: 2089-maximum-matrix-sum/maximum-matrix-sum.rs
impl Solution {
    pub fn max_matrix_sum(matrix: Vec<Vec<i32>>) -> i64 {
        let mut is_negative_even = true;
        let mut smallest_el = i64::MAX;
        let mut sum = 0;
        for row in matrix.iter() {
            for col in row.iter() {
                if *col < 0 {
                    is_negative_even = !is_negative_even;
                }
                let abs = col.abs() as i64;
                sum += abs;
                smallest_el = smallest_el.min(abs);
            }
        }
        if is_negative_even { sum } else { sum - smallest_el * 2 }
    }
}